plugin-server/tests/shared/db.test.ts

```diff
@@ -89,4 +93,82 @@ describe('DB', () => {
+        //  2 | yes          | set      | set             | before                     | yes
+        //  3 | yes          | set_once | set             | before                     | no
+        //  4 | yes          | set      | set_once        | before                     | yes
+        //  5 | yes          | set_once | set_once        | before                     | no
+        //  6 | yes          | set      | set             | equal                      | no
+        //  7 | yes          | set_once | set             | equal                      | no
+        //  8 | yes          | set      | set_once        | equal                      | yes
+        //  9 | yes          | set_once | set_once        | equal                      | no
+        // 10 | yes          | set      | set             | after                      | no
+        // 11 | yes          | set_once | set             | after                      | no
+        // 12 | yes          | set      | set_once        | after                      | yes
+        // 13 | yes          | set_once | set_once        | after                      | yes
+
+        let team: Team
+        let person: Person
+        const uuid = new UUIDT().toString()
+        const distinct_id = 'distinct_id_update_person_properties'
+
+        const FUTURE_TIMESTAMP = DateTime.fromISO('2050-10-14T11:42:06.502Z')
+        const PAST_TIMESTAMP = DateTime.fromISO('2000-10-14T11:42:06.502Z')
+
+        beforeEach(async () => {
+            team = await getFirstTeam(hub)
+            person = await db.createPerson(PAST_TIMESTAMP, {}, team.id, null, false, uuid, [distinct_id])
+        })
+
+        // util to get the new props after an update
+        async function updatePersonProperties(
+            properties: Properties,
+            propertiesOnce: Properties,
+            timestamp: DateTime
+        ): Promise<Person['properties'] | undefined> {
+            await db.updatePersonProperties(team.id, distinct_id, properties, propertiesOnce, timestamp)
+            return (await db.fetchPersonByPersonId(team.id, person.id))?.properties
+        }
+
+        test('update non-existent single property', async () => {
+            const props = await updatePersonProperties({ a: 'a' }, {}, PAST_TIMESTAMP)
+            expect(props).toEqual({ a: 'a' })
+        })
+
+        test('update non-existent property', async () => {
+            const props = await updatePersonProperties({ a: 'a' }, { b: 'b' }, PAST_TIMESTAMP)
+            expect(props).toEqual({ a: 'a', b: 'b' })
+        })
+
+        test('update with newer timestamp - rows [2-5]', async () => {
+            // setup
+            let props = await updatePersonProperties({ r2: 'a', r3: 'b' }, { r4: 'c', r5: 'd' }, PAST_TIMESTAMP)
+            expect(props).toEqual({ r2: 'a', r3: 'b', r4: 'c', r5: 'd' })
+            // update
+            props = await updatePersonProperties({ r2: 'A', r4: 'C' }, { r3: 'B', r5: 'D' }, FUTURE_TIMESTAMP)
```

Comment:
nit: all these letters and numbers are an overload on the readers brain. I rather have a 0 flip to a 1 of a `false` flip to a `true` to indicate an update rather than lowercase -> uppercase

Reply:
the reason I used letters was so we also test that the right key value gets updated, which wouldn't be the case with 0/1 or true/false everywhere, but I could add a comment, would that help or do you think true/false is better still?

Reply:
Well think about this - if I set it all to false and issue updates to true, it'll take me a glance to see which keys got updated, because they will have `true` set.

Just a nit though, so no need to spend a bunch of time on this, and feel free to reject